### DISCUSSION TITLE
Is there a Jupyter notebook for FunASR

### DISCUSSION THREAD

**Discussion by dgoryeo** [modelscope/FunASR#1009] [Q&A] [answered]
Would be very helpful if have a Jupyter notebook to quite proto / test various features.
Also, does one need to use Alibaba cloud, or does FunASR run on any other cloud too?
Thanks

**Comment by LauraGPT** [comment, marked as answer]
Yes. The current repository has a browser-ready Jupyter notebook:

**[Open the FunASR quickstart in Google Colab](https://colab.research.google.com/github/modelscope/FunASR/blob/main/examples/colab/funasr_quickstart.ipynb)**

The same notebook is also available as a normal [GitHub notebook](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/examples/colab/funasr_quickstart.ipynb). It currently covers:

1. installing `funasr`, `modelscope`, and `soundfile`;
2. selecting CPU or GPU automatically;
3. transcribing a public Mandarin sample with Paraformer + FSMN-VAD + punctuation;
4. uploading your own `.wav`, `.mp3`, `.m4a`, or `.flac` file;
5. exporting the result as JSON.

You **do not need an Alibaba Cloud account or Alibaba Cloud compute**. FunASR runs in:

- Google Colab;
- a local Jupyter/JupyterLab environment;
- another cloud provider's VM or notebook;
- a normal Python process on CPU, CUDA GPU, Apple MPS, or other supported hardware.

The default examples may download model files from ModelScope, and the public sample happens to be hosted on Alibaba OSS, but those are artifact/file hosts rather than a requirement to run on Alibaba Cloud. You can replace the sample URL with a local file. FunASR also supports Hugging Face model downloads through `hub="hf"` where the selected model is available, or fully local model paths after downloading the artifacts.

For a local notebook environment:

```bash
python -m pip install jupyterlab funasr modelscope soundfile
jupyter lab
```

Then use the same `AutoModel` cells from the notebook. The repository also exposes the Colab badge directly in its [current Quick Start](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/README.md#L29-L37).
